www/assets/js/components/SearchPage.test.tsx

```diff
@@ -0,0 +1,372 @@
+import { render, screen, fireEvent } from "@testing-library/react"
+import { act } from "react"
+import { search } from "../lib/api"
+import { times } from "ramda"
+import {
+  INITIAL_FACET_STATE,
+  LearningResourceType,
+  serializeSearchParams
+} from "@mitodl/course-search-utils"
+
+import SearchPage from "./SearchPage"
+
+import { DEFAULT_UI_PAGE_SIZE, COMPACT_UI_PAGE_SIZE } from "../lib/constants"
+
+import { makeCourseResult } from "../factories/search"
+import { createMemoryHistory, InitialEntry } from "history"
+
+const mockGetResults = () =>
+  times(() => makeCourseResult(), DEFAULT_UI_PAGE_SIZE).map(result => ({
+    _source: result
+  }))
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let resolver = (_extraData?: any) => {
+  // pass
+}
+
+const resolveSearch = (extraData = {}) =>
+  act(async () => {
+    resolver(extraData)
+  })
+
+jest.mock("../lib/api", () => ({
```

Comment:
I generally think we should mock at the network level (fetch or axios directly, or possibly use something like [msw](https://github.com/mswjs/msw), where fetch/axios don't have to be mocked at all).

That said, mocking the search requests is a pain, and I think this is a perfectly reasonable approach 👍